src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -150,8 +152,15 @@ class MarathonSchedulerService @Inject() (
   override def run(): Unit = {
     log.info("Beginning run")
 
-    // The first thing we do is offer our leadership.
-    electionService.offerLeadership(this)
+    // ElectionService immediately offers leadership. We want to register to listen for when leadership was acquired
+    electionService.leaderTransitionEvents.takeWhile {
```

Comment:
Let's run this on a different dispatcher because it calls blocking stuff. I think we don't want to block the default EC.

Reply:
Blocking calls happen in a different (single threaded) execution context already, upstream.

Reply:
✅ https://github.com/mesosphere/marathon/pull/5525/commits/7437172e1aedc251161975ea74ea1c991a43576c